modules/jupyter/main.tf

```diff
@@ -57,44 +57,40 @@ resource "kubernetes_annotations" "hub" {
     namespace = var.namespace
   }
   annotations = {
-    "iam.gke.io/gcp-service-account" = "${var.workload_identity_service_account}@${var.project_id}.iam.gserviceaccount.com"
+    "iam.gke.io/gcp-service-account" = module.jupyterhub-workload-identity.gcp_service_account_email
   }
   depends_on = [
     helm_release.jupyterhub,
-    module.jupyterhub-workload-identity
   ]
 }
 
-data "google_service_account" "sa" {
-  account_id = var.workload_identity_service_account
-  depends_on = [
-    helm_release.jupyterhub,
-    module.jupyterhub-workload-identity
-  ]
-}
+# data "google_service_account" "sa" {
+#   account_id = var.workload_identity_service_account
+#   depends_on = [
+#     helm_release.jupyterhub,
+#     module.jupyterhub-workload-identity
+#   ]
+# }
 
 resource "google_service_account_iam_binding" "hub-workload-identity-user" {
   count              = var.add_auth ? 1 : 0
-  service_account_id = data.google_service_account.sa.name
+  service_account_id = module.jupyterhub-workload-identity.gcp_service_account
   role               = "roles/iam.workloadIdentityUser"
 
   members = [
     "serviceAccount:${var.project_id}.svc.id.goog[${var.namespace}/hub]",
     "serviceAccount:${var.project_id}.svc.id.goog[${var.namespace}/${var.workload_identity_service_account}]",
   ]
   depends_on = [
-    helm_release.jupyterhub,
-    module.jupyterhub-workload-identity
```

Comment:
why remove the depend on?

Reply:
https://github.com/GoogleCloudPlatform/ai-on-gke/pull/323/files#diff-4d5d069574473acf60a9166857427b2a70bfbf98a084d797678f4268b07cf3b3R69

this line implicitly make workload-identity module depends on